tests/StateSnapshot.test.js

```diff
@@ -14,307 +14,352 @@ const ruleTester = new RuleTester({
-
-      useEffect(() => {
-        if (c === 'a1c') {
-          state.a1.b.c = 'example';
-        }
-      }, [c]);
-    }`,
+       function useExample2(s) {
+         const {b: {c} } = useSnapshot(s.a1);
+    
+         useEffect(() => {
+           if (c === 'a1c') {
+             state.a1.b.c = 'example';
+           }
+         }, [c]);
+       }`,
 
     `
-    function useProxyStateExample(a) {
-      const s = useSnapshot(a);
-
-      useEffect(() => {
-          a.a = 'example';
-      }, [s.a]);
-    }`,
+       function useProxyStateExample(a) {
+         const s = useSnapshot(a);
+    
+         useEffect(() => {
+             a.a = 'example';
+         }, [s.a]);
+       }`,
     `const state = proxy({ count: 0 });
-    function App() {
-      const snap = useSnapshot(state);
-      const handleClick = useCallback(() => {
-        console.log(snap.count);
-      }, [snap.count]);
-      return (
-        <div>
-          {snap.count} <button onClick={handleClick}>click</button>
-        </div>
-      );
+       function App() {
+         const snap = useSnapshot(state);
+         const handleClick = useCallback(() => {
+           console.log(snap.count);
+         }, [snap.count]);
+         return (
+           <div>
+             {snap.count} <button onClick={handleClick}>click</button>
+           </div>
+         );
+       }
+       `,
+    `
+       const DirectReadComponent = () => {
+         const debounceSnap = useSnapshot(state);
+         return <div>{debounceSnap}</div>;
+       };
+       `,
+    `
+       const ObjectPatternReadOne = () => {
+         const {b: {c} } = useSnapshot(state);
+         return <div>{c}</div>;
+       };`,
+    `
+       const ObjectPatternReadTwo = () => {
+         const {c} = useSnapshot(state);
+         return <div>{c}</div>;
+       };`,
+    `
+     const state = proxy({ count: 1 })
+     const useDoubled = () => {
+       const snap = useSnapshot(state)
+       return snap.count * 2
+     }
+     const Component = ()=> {
+       const doubled = useDoubled()
+       return <>{doubled}</>
+     }
+      `,
+    `
+    const Component = ()=>{
+      const { index } = useSnapshot(idxStore);
+    React.useEffect(() => {
+     const getUsers = async (id) => {
+       setUsers()
+     };
+     getUsers(index);
+  }, [index])
+  return <></>
     }
-    `,
+`,
     `
-    const DirectReadComponent = () => {
-      const debounceSnap = useSnapshot(state);
-      return <div>{debounceSnap}</div>;
-    };
-    `,
+	const state = proxy({count:1});
+	const useDoubled = () => {
+		const snap = useSnapshot(state)
+		const doubled = snap.count*2;
+		return doubled 
+	}
+	`,
     `
-    const ObjectPatternReadOne = () => {
-      const {b: {c} } = useSnapshot(state);
-      return <div>{c}</div>;
-    };`,
+	const state = proxy({count:1});
+	const useDoubled = () => {
+		const snap = useSnapshot(state)
+		const {doubled} = {doubled:snap.count*2};
+		return doubled 
+	}
+	`,
```

Comment:
```suggestion
	`,
	`
	const useDoubled = ({ state }) => {
		const snap = useSnapshot(state)
		const {doubled} = {doubled:snap.count*2};
		return doubled 
	}
	`,
```
How about this?

Reply:
this should work no issues, we use the `useSnapshot` definition to find out what a snapshot is so it should work fine

Reply:
added to the commit anyway